samples/app/src/main/java/com/example/app/storage/CustomStorageWithFallback.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.example.app.storage
+
+import android.content.Context
+import kotlinx.serialization.Serializable
+import org.forgerock.android.auth.AccessToken
+import org.forgerock.android.auth.SSOToken
+import org.forgerock.android.auth.TokeStorage
+import org.forgerock.android.auth.storage.CookiesStorage
+import org.forgerock.android.auth.storage.SSOTokenStorage
+import org.forgerock.android.auth.storage.MemoryStorage
+import org.forgerock.android.auth.storage.Storage
+
+/**
+ * A custom storage class with a fallback mechanism.
+ *
+ * @param T The type of object to be stored.
+ * @param context The application context.
+ * @param flag A flag used to control the storage type.
+ * @param storage The primary storage implementation.
+ * @param fallback The fallback storage implementation.
+ */
+class CustomStorageWithFallback<T : @Serializable Any>(
+    private val context: Context,
+    private val flag: String,
+    storage: Storage<T>,
+    private val fallback: Storage<T>
+) : Storage<T> {
+
+    @Volatile
+    private var current: Storage<T> = storage
+
+    /**
+     * Save an item to the current storage. If an error occurs, switch to the fallback storage.
+     *
+     * @param item The item to be saved.
+     */
+    override fun save(item: T) {
+        try {
+            // Save the item to the current storage.
+            current.save(item)
+        } catch (e: Throwable) {
+            // If an error occurs, switch to the fallback storage.
+            context.getSharedPreferences("storage-control", Context.MODE_PRIVATE).edit()
+                .putInt(flag, 1).apply()
+            fallback.save(item)
+            current = fallback
+        }
+    }
+
+    /**
+     * Retrieve an item from the current storage.
+     *
+     * @return The retrieved item, or null if no item is found.
+     */
+    override fun get(): T? {
+        return current.get()
+    }
+
+    /**
+     * Delete an item from the current storage.
+     */
+    override fun delete() {
+        current.delete()
+    }
+}
+
+/**
+ * Load the SSO token storage with a fallback mechanism.
+ *
+ * @param context The application context.
+ * @return The storage instance for SSO tokens.
+ */
+fun loadSSOTokenStorage(context: Context): Storage<SSOToken> {
+    return loadStorage(
+        context,
+        "ssoStorage",
+        { SSOTokenStorage(context) },
+        { MemoryStorage() }
+    )
+}
+
+/**
+ * Load the token storage with a fallback mechanism.
+ *
+ * @param context The application context.
+ * @return The storage instance for tokens.
+ */
+fun loadTokenStorage(context: Context): Storage<AccessToken> {
+    return loadStorage(
+        context,
+        "tokenStorage",
+        { TokeStorage(context) },
```

Comment:
Should this be `TokenStorage(context)`, i.e. is it missing the `n`?

Reply:
Updated.